File: software/examples/ruby/example_threshold.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_current12'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change to your UID

ipcon = IPConnection.new # Create IP connection
c12 = BrickletCurrent12.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Get threshold callbacks with a debounce time of 10 seconds (10000ms)
c12.set_debounce_period 10000

# Register threshold reached callback for current greater than 5 A (parameter has unit mA)
c12.register_callback(BrickletCurrent12::CALLBACK_CURRENT_REACHED) do |current|
  puts "Current: #{current/1000.0} A"
end

# Configure threshold for "greater than 5 A" (unit is mA)
c12.set_current_callback_threshold '>', 5*1000, 0

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect

File: software/examples/javascript/ExampleThreshold.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change to your UID

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var c = new Tinkerforge.BrickletCurrent12(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Get threshold callbacks with a debounce time of 10 seconds (10000ms)
        c.setDebouncePeriod(10000);

        // Configure threshold for current "greater than 5 A" (unit is mA)
        c.setCurrentCallbackThreshold('>', 5*1000, 0);
    }
);

// Register current reached callback
c.on(Tinkerforge.BrickletCurrent12.CALLBACK_CURRENT_REACHED,
    // Callback function for current reached callback (parameter has unit mA)
    function (current) {
        console.log('Current: ' + current/1000.0 + ' A');
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);
